pkg/util/iptables/iptables_test.go

```diff
@@ -771,3 +771,224 @@ func TestReload(t *testing.T) {
 		t.Errorf("wrong CombinedOutput() log, got %s", fcmd.CombinedOutputLog[6])
 	}
 }
+
+func TestSave(t *testing.T) {
+	output := `# Generated by iptables-save v1.6.0 on Thu Jan 19 11:38:09 2017
+*filter
+:INPUT ACCEPT [15079:38410730]
+:FORWARD ACCEPT [0:0]
+:OUTPUT ACCEPT [11045:521562]
+COMMIT
+# Completed on Thu Jan 19 11:38:09 2017`
+
+	fcmd := exec.FakeCmd{
+		CombinedOutputScript: []exec.FakeCombinedOutputAction{
+			// iptables version check
+			func() ([]byte, error) { return []byte("iptables v1.9.22"), nil },
+			func() ([]byte, error) { return []byte(output), nil },
+			func() ([]byte, error) { return nil, &exec.FakeExitError{Status: 1} },
+		},
+	}
+	fexec := exec.FakeExec{
+		CommandScript: []exec.FakeCommandAction{
+			func(cmd string, args ...string) exec.Cmd { return exec.InitFakeCmd(&fcmd, cmd, args...) },
+			func(cmd string, args ...string) exec.Cmd { return exec.InitFakeCmd(&fcmd, cmd, args...) },
+			func(cmd string, args ...string) exec.Cmd { return exec.InitFakeCmd(&fcmd, cmd, args...) },
+		},
+	}
+	runner := New(&fexec, dbus.NewFake(nil, nil), ProtocolIpv4)
+	defer runner.Destroy()
+	// Success.
+	o, err := runner.Save(TableNAT)
+	if err != nil {
+		t.Errorf("expected success, got %v", err)
```

Comment:
`t.Errorf` should be `t.Fatalf` because `o` may be nil. `TestSaveAll` has the same issue.